### PR TITLE
Add build_even/odd types to split Arduino tests

### PR DESCRIPTION
**Pull request by earlephilhower**
Arduino builds are now taking 49 minutes or more, and failing due to
Travis timeouts.

Split the Arduino build task into even and odd halves, where each job
will build every other test.  This will reduce any test's runtime
by 50% and speed up checks to avoid the timeout.

### CODE REVIEW

**Comment by earlephilhower** [comment]
Just verified that the combination of both _odd and _even run the same tests as the single combined run (96 in total now).

**Comment by earlephilhower** [comment]
Added @igrr to get his input, too.  Right now it's looking like jobs fail almost all the time due to timeouts, so if we want to get any merges in, this (or some other solution) has probably got to be merged first.

**Comment by Juppit** [comment]
You can use travis [build stages](https://docs.travis-ci.com/user/build-stages/), all stages are executed consecutively, and each stage can run up to one hour.

File: tests/common.sh [hunk 1]
```diff
@@ -239,7 +249,13 @@ fi
...
     install_arduino
-    build_sketches_with_arduino
+    build_sketches_with_arduino 1 0
+elif [ "$BUILD_TYPE" = "build_even" ]; then
+    install_arduino
+    build_sketches_with_arduino 2 0
+elif [ "$BUILD_TYPE" = "build_odd" ]; then
+    install_arduino
+    build_sketches_with_arduino 2 1
 elif [ "$BUILD_TYPE" = "platformio" ]; then
```
[the remark this answers is not in the document]
Comment (earlephilhower) [reply tests/common.sh]: It's not needed.  PlatformIO builds the examples 2x (yes, two times) as fast as Arduino.  So it takes ~28mins today vs 50+mins for Ardunio.  Should we go from 96 to 200 examples, I'll revisit this statement, but for now we're OK. :smile:

[the remark this answers is not in the document]
Comment (devyte) [reply tests/common.sh]: Looking at the.times, you're right. Even with load I see the platformio runtime doesn't go above 40mins.

**Comment by devyte** [comment]
Merging this now in the interest of moving forward.

**Pull request merged**